oslabs-beta/DockerLocal
# Leave one out fix

## Description:
- Previous Leave-one-out had assumed container name in dockerController would match the checked repositories and also contained an incorrect conditional statement. This was a false assumption since they do not match. I used buildPath to grab correct repositories names that are currently stored in active Project. This allows only checked repositories that have been cloned locally to be included in Docker compose file.
- Additionally, I added logic so that docker compose yaml file in project root folder will not have to be deleted before using the compose button again. The same filePath can be used over and over agin and will display the correct compose file according to the user's last specifications on clicking compose File button
## Changes I Made:
- `dockerCompose.tsx:`
   - `dockerCompose.createDockerCompose:`
       - remove checking if file exists before creating docker compose file with write file to minimize our local storage and refrain from adding unnecessary delete file logic
       - Change 'checked' repositories logic before appending to file so that only already cloned repos will appear in compose file shown to user. (leave-one-out functionality)

## How to Test:
- npm install
- npm run start
- Check that compose button only creates services with repos that are cloned:
   - Add Repos at least two repos with dockerfiles
   - Clone only one of those repos with dockerfiles
   - Use compose button
     - Notice that only the cloned repo that contains dockerfiles appears
- Check that you can leave one out:
   - Create new project
   - Clone at least two repos with dockerfiles
   - Choose to compose using only of of those repos
   - Verify leave one out functionality
- Check that compose files does not need to be deleted before clicking the compose button again
   - In either of the previous projects that you have created, check a few repos (at least on with dockerfiles)
   - Click compose File button
   - Notice that the repo is not added to the bottom like before but is a fresh new file that has the currently clicked on repos with the same filePath as before.
